PlayCover/Utils/URLHandler.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  URIHandler.swift
+//  PlayCover
+//
+//  Created by Venti on 14/02/2023.
+//
+
+import Foundation
+
+enum URLTypes: Int, Equatable {
+    case source
+    case keymap
+    case app
+}
+
+enum URLAction: Int, Equatable {
+    case add
+    case remove
+    case update
+    case install
+    case open
+}
+
+class URLObservable: ObservableObject {
+    @Published var url: String?
+    @Published var type: URLTypes?
+    @Published var action: URLAction?
+
+    public static var shared = URLObservable()
+}
+
+struct URLHandler {
+    public static var shared = URLHandler()
+
+    func processURL(url: URL) {
+        guard let urlComponenents = NSURLComponents(url: url, resolvingAgainstBaseURL: false),
+            let uriPath = urlComponenents.path,
+            let params = urlComponenents.queryItems else {
+                // Fall back to old url handler (for files)
+                if url.pathExtension == "ipa" {
+                    Installer.install(ipaUrl: url, export: false, returnCompletion: { _ in
+                    Task { @MainActor in
+                        AppsVM.shared.fetchApps()
+                        NotifyService.shared.notify(
+                            NSLocalizedString("notification.appInstalled", comment: ""),
+                            NSLocalizedString("notification.appInstalled.message", comment: "")
+                        )
+                    }})
+                }
+                return
+            }
+        // URI format: playcoverapp://<object>?action=<action>&<param>=<value>
+        // Example: playcoverapp://source?action=add&url=https://homebrew.playcover.io
+
+        // Switch case for main uri path
+        switch uriPath {
+        case "/source":
+            processSourceURL(params: params)
+        default:
+            // Print URL to console and break
+            print("Unknown URL: \(url)")
+        }
+    }
+
+    func processSourceURL(params: [URLQueryItem]) {
+        switch params[0].value {
+        case "add":
+            // Add source
+            if let url = params[1].value {
```

Comment:
There is missing error handling if the request for actions includes no parameters. For instance, calling `playcover:///source?action=add` will crash the app due to an index out of range error.

Reply:
A similar situation occurs on the switch of line 66 if `playcover:///source?` is queried.

Reply:
Added the necessary checks

Reply:
The crashes still happen on https://github.com/PlayCover/PlayCover/pull/538/commits/9db78398d4079f087deb9148bc25ae66f9d48cc5. I propose merging the following changes to solve the issue https://github.com/Depal1/PlayCover/compare/9db7839...804016c.

Reply:
Should be fine now. Added one more check to make sure the params is what we expect